### PR TITLE
Reworks dashboard

### PR DESCRIPTION
**Pull request by DominikAngerer**
# Reworked Dashboard and Learn Area

The reworked dashboard got rid of handlebars and uses Vue.js to add reactivity to the existing template. Recreated style using Tailwind for better style. Logic slightly touched to get rid of handlebars rendering but still keeps data loading from `dashboard.json` which will be generated by python.

image

image

image

## How to access

In the base directory `kuegiBot` not `kuegiBot/dashboard/` execute the following command:

```
python3 -m http.server 8000
```

### Available routes

- Dashboard: http://localhost:8001/dashboard/index.html
- Learn: http://localhost:8001/dashboard/learn.html

### CODE REVIEW

File: dashboard/index.html [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+                        </template>
+                    </div>
+                </div>
+            </div>
+        </main>
+    </div>
+
+
+    <script>
+        const Dashboard = {
```
Comment (kuegi) [inline dashboard/index.html:110]: do we need this as part of the html or can we extract it into a .js file?

File: dashboard/index.html [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+                },
+            },
+            mounted() {
+                this.viewInitialized = true
+                this.loadDashboardData()
+            },
+            methods: {
+                loadDashboardData() {
+                    this.dashboard.isLoading = true
+                    window.fetch('/dashboard/dashboard.json?v=' + Date.now())
```
Comment (kuegi) [inline dashboard/index.html:135]: would prefer to keep it on the same level as the index.html or at least relative to the index.html and not absolute.
 makes it more flexible to be used on different production-server setups

**Review by kuegi — COMMENTED** [review]
Whats the benefit of having the used libs within the repo as compared to load it from a public cdn?

File: dashboard/index.html (hunk 1 above)
Comment (DominikAngerer) [reply dashboard/index.html:110]: We can extract it into a JS file. Tried to keep it simple so the dynamic template on the top can be developed alongside the logic that is used to render it - Basically "SFC / Single Page Components" https://vuejs.org/v2/guide/single-file-components.html without the need of having a build system :)

File: dashboard/index.html (hunk 2 above)
Comment (DominikAngerer) [reply dashboard/index.html:135]: No problem with that - will update.

**Comment by DominikAngerer** [comment]
The benefit of not having third party scripts that are not in control accessing your data. Normally this would be done using a build tool and bundling but the keep it simple we just keep the version we want and host it ourselves. This will also speed it up as no additional DNS resolving needs to happen :)

**Comment by kuegi** [comment]
@ support : PR can be deleted.

**Pull request closed without merging**